xarray/core/groupby.py

```diff
@@ -279,7 +279,7 @@ def _yield_binary_applied(self, func, other):
                 raise TypeError('GroupBy objects only support binary ops '
                                 'when the other argument is a Dataset or '
                                 'DataArray')
-            except KeyError:
+            except (KeyError, ValueError):
```

Comment:
`indexing.get_dim_indexers()` checks for invalid indexers and raises a `ValueError` if any (https://github.com/benbovy/xarray/blob/52b88d8660e2af7ab823954009b913e0781e9bfd/xarray/core/indexing.py#L228)

I set a `ValueError` given similar checks that I've seen in `Dataset.isel()` and `Variable.isel()`. I don't know if a `KeyError` would be more appropriate?